frontend/packages/topology/src/data-transforms/DataModelProvider.tsx

```diff
@@ -18,13 +23,23 @@ const DataModelProvider: React.FC<DataModelProviderProps> = ({ namespace, childr
   }, [namespace]);
 
   const modelFactories = useExtensions<TopologyDataModelFactory>(isTopologyDataModelFactory);
+  const [dynamicModelFactories, isModelResolved] = useResolvedExtensions<
+    DynamicTopologyDataModelFactory
+  >(isDynamicTopologyDataModelFactory);
```

Comment:
Can both paths use `useResolvedExtensions` so that we don't have one path doing manual resolution.